policyengine_us/variables/gov/states/ny/tax/income/ny_supplemental_tax.py

```diff
@@ -53,20 +53,14 @@ def formula(tax_unit, period, parameters):
             applicable_amount / p.phase_in_length,
         )
 
-        # edge case for high agi
-        agi_limit = select(
-            in_each_status,
-            [
-                single.thresholds[-1],
-                joint.thresholds[-1],
-                hoh.thresholds[-1],
-                surviving_spouse.thresholds[-1],
-                separate.thresholds[-1],
-            ],
-        )
+        # For AGI above the high threshold, apply flat top rate to all income
+        high_agi_threshold = p.high_agi_threshold
+        # Create array for marginal_rates lookup
+        # Use +100 for float32 precision at $25M threshold
+        high_agi_lookup = ny_agi * 0 + high_agi_threshold + 100
```

Comment:
Did the original logic not work? I wonder if need a new parameter for this if we have it as part of the scale parameter which already exists